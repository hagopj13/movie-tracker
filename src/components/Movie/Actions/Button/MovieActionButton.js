// @flow
import React from 'react';
import type { Node } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Tooltip from '@material-ui/core/Tooltip';
import Fab from '@material-ui/core/Fab';
import { lightBlue } from '@material-ui/core/colors';

type Props = {
  children: Node,
  title: string,
  onClick: () => void,
};

const useStyles = makeStyles((theme) => ({
  tooltip: {
    fontSize: theme.typography.subtitle2.fontSize,
  },
  fab: {
    backgroundColor: lightBlue[800],
    '&:hover': {
      backgroundColor: lightBlue[900],
    },
  },
}));

const MovieActionButton = (props: Props, ref) => {
  const { children, title, onClick } = props;

  const classes = useStyles();

  return (
    <Tooltip classes={{ tooltip: classes.tooltip }} title={title}>
      <Fab className={classes.fab} ref={ref} size="medium" onClick={onClick}>
        {children}
      </Fab>
    </Tooltip>
  );
};

export default React.forwardRef<Props, Fab>(MovieActionButton);
