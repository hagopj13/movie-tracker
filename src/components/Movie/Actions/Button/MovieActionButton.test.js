import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Fab from '@material-ui/core/Fab';
import Tooltip from '@material-ui/core/Tooltip';

import MovieActionButton from './MovieActionButton';

describe('MovieActionButton component', () => {
  let shallow;
  let title;
  let wrapper;

  const mockClick = jest.fn();
  const ChildComponent = () => <div>Some child component</div>;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    jest.clearAllMocks();
    title = 'Some title';
    const mockProps = {
      title,
      onClick: mockClick,
    };
    wrapper = shallow(
      <MovieActionButton {...mockProps}>
        <ChildComponent />
      </MovieActionButton>,
    );
  });

  it('should render the MovieActionButton component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the child component', () => {
    expect(wrapper.exists(ChildComponent)).toBe(true);
  });

  it('should pass the title to the Tooltip', () => {
    expect(wrapper.find(Tooltip).prop('title')).toBe(title);
  });

  it('should call onClick when Fab is clicked', () => {
    wrapper.find(Fab).simulate('click');
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
});
