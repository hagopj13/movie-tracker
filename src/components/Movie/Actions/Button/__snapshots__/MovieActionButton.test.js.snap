// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieActionButton component should render the MovieActionButton component correctly 1`] = `
<WithStyles(ForwardRef(Tooltip))
  classes={
    Object {
      "tooltip": "makeStyles-tooltip-1",
    }
  }
  title="Some title"
>
  <WithStyles(ForwardRef(Fab))
    className="makeStyles-fab-2"
    onClick={[MockFunction]}
    size="medium"
  >
    <ChildComponent />
  </WithStyles(ForwardRef(Fab))>
</WithStyles(ForwardRef(Tooltip))>
`;
